index.html

```diff
@@ -705,31 +710,65 @@
                                 var inputPort, outputPort, controlPort, commandPort;
                                 if (cellvar.model.in.height == null) {
                                     inputPort = 0;
-                                } else {
+                                }
+                                else {
                                     inputPort = cellvar.model.in.height;
                                 }
                                 if (cellvar.model.out.height == null) {
                                     outputPort = 0;
-                                } else {
+                                }
+                                else {
                                     outputPort = cellvar.model.out.height;
                                 }
                                 if (cellvar.model.evtin.height == null) {
                                     controlPort = 0;
-                                } else {
+                                }
+                                else {
                                     controlPort = cellvar.model.evtin.height;
                                 }
                                 if (cellvar.model.evtout.height == null) {
                                     commandPort = 0;
-                                } else {
+                                }
+                                else {
                                     commandPort = cellvar.model.evtout.height;
                                 }
+
+                                var flip = false, mirror = false;
+                                if(cell.style.indexOf('stencilFlipV') == -1){
+                                    flip = false;
+                                }
+                                else
+                                {
+                                    var x = cell.style.indexOf('stencilFlipV');
+                                    if(cell.style.charAt(x+13) == 0){
```

Comment:
We need it when stencilV is not present in style.. so that we add it with
value 1. And when it is present we need to check it is 0 or 1.
On 12 Jul 2016 3:18 p.m., "Adhitya Kamakshidasan" <notifications@github.com>
wrote:

> In index.html
> <https://github.com/FOSSEE/xcos-on-web/pull/170#discussion_r70408085>:
>
> >                                      commandPort = cellvar.model.evtout.height;
> >                                  }
> > +
> > +                                var flip = false, mirror = false;
> > +                                if(cell.style.indexOf('stencilFlipV') == -1){
> > +                                    flip = false;
> > +                                }
> > +                                else
> > +                                {
> > +                                    var x = cell.style.indexOf('stencilFlipV');
> > +                                    if(cell.style.charAt(x+13) == 0){
>
> Why do we need this condition? Is there a better way of doing this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FOSSEE/xcos-on-web/pull/170/files/b06ae980294c811c105c388c011235008ec6b2e4#r70408085>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALzP2myBAJHzc6xBVWtbfmvp-Z4fsqHTks5qU2MEgaJpZM4JKJM6>
> .
>